src/common/stats/StatsManager.h

```diff
@@ -13,10 +13,48 @@
 #include <folly/RWSpinLock.h>
 #include <folly/stats/MultiLevelTimeSeries.h>
 #include <folly/stats/TimeseriesHistogram.h>
+#include <gtest/gtest_prod.h>
 
 namespace nebula {
 namespace stats {
 
+class StatsManager;
+
+class MetricsSerializer {
+public:
+    virtual ~MetricsSerializer() = default;
+    std::string serialize(StatsManager& sm) const {
+        std::ostringstream ss;
+        serialize(ss, sm);
+        return ss.str();
+    }
+    virtual void serialize(std::ostream& out, StatsManager& sm) const = 0;
+};
+
+class PrometheusSerializer : public MetricsSerializer {
+public:
+    virtual ~PrometheusSerializer() = default;
+    void serialize(std::ostream& out, StatsManager& sm) const override {
+        prometheusSerialize(out, sm);
+    }
+    void prometheusSerialize(std::ostream& out, StatsManager& sm) const;
+    struct Label {
+        std::string name;
+        std::string value;
+    };
+    void annotate(std::ostream& out, const std::string& metricName,
+        const std::string& metricType) const;
+    void writeValue(std::ostream& out, double value) const;
+    void writeValue(std::ostream& out, const int64_t value) const;
+    void writeValue(std::ostream& out, const std::string& value) const;
+    template <typename T = std::string>
+    void writeHead(std::ostream& out, const std::string& family,
+        const std::vector<Label>& labels, const std::string& suffix = "",
+        const std::string& extraLabelName = "",
+        const T extraLabelValue = T()) const;
+    void writeTail(std::ostream& out, const std::int64_t timestampMs = 0) const;
+};
```

Comment:
These serializer classes are preferable defined in another new source files.

Reply:
Yes, I think so.